aiourlshortener/shorteners/base.py

```diff
@@ -49,5 +50,14 @@ def expand(self, url: str) -> str:
     def close(self):
         if self._session is not None:
             yield from self._session.close()
-        except TypeError:
-            pass
+
+    def __del__(self):
+        if self._session is not None and not self._session.closed:
+            self._session.close()
+
+    @classmethod
+    def __subclasshook__(cls, c):
+        if cls is BaseShortener:
+            if all(hasattr(c, name) for name in ('short', 'expand')):
```

Comment:
@blikenoother The custom `__subclasshook__` is not necessary.

Reply:
@das7pad will remove in version 0.0.4
I am also planning to add `testcases` with `Travis CI` if you can help me identify what testcases should be there, it will be great help

Reply:
@blikenoother I started to work on a PR for CI.
I am currently working with `Jenkins` locally, but as the tests are started via `make` -> `pytest`, they should run with minimal customization in `Travis CI` as well.

Let's move the discussion into a separate issue.